### PR TITLE
fix(kumactl): kumactl -o yaml breaks Mesh* policies

### PR DESCRIPTION
**Pull request by nsikorskiy**
After the release of the new Mesh* policies, fixing issue #679 broke the output of content in YAML format. The fix for issue #679, which involves splitting the policy into a header and spec body and outputting them separately, resulted in the loss of the spec in this particular case. This issue can be resolved by using an alternative method for converting YAML to JSON.

### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Comment by nsikorskiy** [comment]
With fix:
```
type: MeshRetry
mesh: stagings
name: retry-all-stagings
creationTime: "2024-02-15T08:18:20.140897Z"
modificationTime: "2024-02-15T08:18:20.140897Z"
spec:
  targetRef:
    kind: Mesh
  to:
  - targetRef:
      kind: Mesh
    default:
      tcp:
        maxConnectAttempt: 5
      http:
        numRetries: 5
        perTryTimeout: 16s
        backOff:
          baseInterval: 25ms
          maxInterval: 250ms
      grpc:
        numRetries: 5
        perTryTimeout: 16s
        backOff:
          baseInterval: 25ms
          maxInterval: 250ms
```


kumactl:2.6.1
```
creationTime: "2024-02-15T08:18:20.140897Z"
mesh: stagings
modificationTime: "2024-02-15T08:18:20.140897Z"
name: retry-all-stagings
type: MeshRetry
targetRef:
  kind: Mesh
to:
- default:
    grpc:
      backOff:
        baseInterval: 25ms
        maxInterval: 250ms
      numRetries: 5
      perTryTimeout: 16s
    http:
      backOff:
        baseInterval: 25ms
        maxInterval: 250ms
      numRetries: 5
      perTryTimeout: 16s
    tcp:
      maxConnectAttempt: 5
  targetRef:
    kind: Mesh
```

File: app/kumactl/pkg/output/yaml/printer.go [hunk 1]
```diff
@@ -1,14 +1,12 @@
 package yaml
 
 import (
-	"bytes"
 	"io"
 
-	"sigs.k8s.io/yaml"
+	"encoding/json"
+	"gopkg.in/yaml.v2"
```
Comment (slonka) [inline app/kumactl/pkg/output/yaml/printer.go:7]: this needs to be explicitly defined in `go.mod` I think

**Comment by slonka** [comment]
Hi, thanks for contributing. I'm looking into this because it might turn out to be a bigger fix than just one file.

File: app/kumactl/pkg/output/yaml/printer.go (hunk 1 above)
Comment (nsikorskiy) [reply app/kumactl/pkg/output/yaml/printer.go:7]: Thank you. Added to explicit dependencies in go.mod

**Comment by slonka** [comment]
@nsikorskiy can you work on the linter and also resolve conflicts?

**Comment by nsikorskiy** [comment]
> @nsikorskiy can you work on the linter and also resolve conflicts?

I resolved the conflict and added //nolint for module

File: go.mod [hunk 2]
```diff
@@ -76,6 +76,7 @@ require (
 	google.golang.org/grpc v1.62.1
 	google.golang.org/protobuf v1.33.0
 	gopkg.in/natefinch/lumberjack.v2 v2.2.1
+	gopkg.in/yaml.v2 v2.4.0
```
Comment (lahabana) [inline go.mod:79]: I think we shouldn't be using this anymore

Comment (nsikorskiy) [reply go.mod:79]: I used gopkg.in/yaml.v2 to preserve the sequence of fields. I couldn't find a way to do this with sigs.k8s.io. 
Additionally, sigs.k8s.io itself uses gopkg.in/yaml.v2 internally. I believe that here you can use this library directly.

Comment (nsikorskiy) [reply go.mod:79]: Okay, I replaced the explicit use of gopkg.in/yaml.v2  with sigs.k8s.io/yaml/goyaml.v2

**Comment by lahabana** [comment]
#9779 is the correct fix for this. Thanks for pointing this out @nsikorskiy !

**Pull request closed without merging**